hr_holidays_public/models/hr_holidays_public.py

```diff
@@ -172,3 +173,43 @@ def _check_date_state_one(self):
                 _("You can't create duplicate public holiday per date %s.") % self.date
             )
         return True
+
+    def _prepare_holidays_meeting_values(self):
+        self.ensure_one()
+        categ_id = self.env.ref("hr_holidays_public.event_type_holiday", False)
+        meeting_values = {
+            "name": (
+                "{} ({})".format(self.name, self.year_id.country_id.name)
+                if self.year_id.country_id
+                else self.name
+            ),
+            "description": ", ".join(self.state_ids.mapped("name")),
+            "categ_ids": [(6, 0, categ_id.ids if categ_id else [])],
+            "start": self.date,
+            "stop": self.date,
+            "allday": True,
+            "partner_ids": False,
+            "user_id": SUPERUSER_ID,
+            "state": "open",
+            "privacy": "confidential",
+            "show_as": "busy",
+        }
+        return meeting_values
+
+    @api.constrains("date", "name", "year_id", "state_ids")
+    def _update_calendar_event(self):
```

Comment:
why this method is needed? when a public line is edited, why resetting its meeting? I mean, you are changing almost all of the meeting fields...

Reply:
This is to keep the calendar event synchronized with the public holiday.
What's the use case you're worried about? The meeting is not meant to be edited by anyone, anyways

Reply:
@ivantodorovich Why this is a constrains method ? why not onchange ?

Reply:
mmm is it because the write method is called?

Reply:
onchanges are not triggered when creating/updating records programatically, you'd need to manually call the _onchange

constrains, on the other hand, will always be triggered if one of the fields changes